improver/calibration/reliability_calibration.py

```diff
@@ -1268,10 +1268,19 @@ def _interpolate(
 
         forecast_probabilities = np.ma.getdata(forecast_threshold).flatten()
 
+        # interpolate using scipy first to get extrapolated values at endpoints
         interpolation_function = scipy.interpolate.interp1d(
             reliability_probabilities, observation_frequencies, fill_value="extrapolate"
         )
-        interpolated = interpolation_function(forecast_probabilities.data)
+        y_0, y_1 = interpolation_function([0, 1])
+        xp = np.copy(reliability_probabilities)
+        xp[0] = 0
+        xp[-1] = 1
+        fp = np.copy(observation_frequencies)
+        fp[0] = y_0
+        fp[-1] = y_1
+
+        interpolated = np.interp(forecast_probabilities.data, xp, fp)
```

Comment:
I has a look at the numpy and scipy API for these functions and the scipy interp1d has other interpolation kinds (such as nearest or cubic), where numpy is linear only. Given we are using linear, this is perfect, however do you think it would be worth a comment that other interpolation options would require use of the scipy method?

Reply:
Done